src/Sylius/Component/Core/Cart/Context/ShopBasedCartContext.php

```diff
@@ -31,12 +32,22 @@ final class ShopBasedCartContext implements CartContextInterface
 
     private ShopperContextInterface $shopperContext;
 
+    private ?TokenStorageInterface $tokenStorage;
+
     private ?OrderInterface $cart = null;
 
-    public function __construct(CartContextInterface $cartContext, ShopperContextInterface $shopperContext)
-    {
+    public function __construct(
+        CartContextInterface $cartContext,
+        ShopperContextInterface $shopperContext,
+        ?TokenStorageInterface $tokenStorage = null
+    ) {
         $this->cartContext = $cartContext;
         $this->shopperContext = $shopperContext;
+        $this->tokenStorage = $tokenStorage;
+
+        if ($tokenStorage === null) {
+            @trigger_error('Not passing tokenStorage through constructor is deprecated in Sylius 1.10 and it will be removed in Sylius 2.0');
```

Comment:
```suggestion
            @trigger_error('Not passing tokenStorage through constructor is deprecated in Sylius 1.10.9 and it will be removed in Sylius 2.0');
```

Reply:
This deprecation should be mentioned in upgrade files, both 1.10 and 1.11 IMO